### PR TITLE
Remove octal escape literals

### PR DESCRIPTION
**Pull request by tanishiking**
Closes https://github.com/lampepfl/dotty/issues/10895
Ref https://github.com/scala/scala/pull/6324

This PR removes octal escape in String literals, which was deprecated and removed in scala 2.

### CODE REVIEW

File: tests/neg/t7292-removal.scala [hunk 1]
```diff
@@ -0,0 +1,5 @@
+object OctalEscapes {
+  val chr1 = '\0' // error
+  val str1 = "abc\123456" // error
+  val lf = '\012' // error
+}
```
Comment (tanishiking) [inline tests/neg/t7292-removal.scala:5]: Moved from `untried`

File: compiler/src/dotty/tools/dotc/parsing/Scanners.scala [hunk 2]
```diff
@@ -1157,6 +1158,8 @@ object Scanners {
               nextChar()
             }
           }
+          val alt = if (oct == LF) "\\n" else "\\u%04x" format oct
```
Comment (som-snytt) [inline compiler/src/dotty/tools/dotc/parsing/Scanners.scala:1161]: `if oct == LF then raw"\n" else f"\u$oct%04x"` for style points?

Comment (tanishiking) [reply compiler/src/dotty/tools/dotc/parsing/Scanners.scala]: 👍 fixed

**Review by odersky — APPROVED** [review]
Thanks for the fix!

**Pull request merged**